File: PetShop-petopia/PetShop-petopia-serverr/src/main/java/com/pet/entity/Notification.java
package com.pet.entity;


import com.pet.enums.NotificationType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Notification {
    @Id
    @EqualsAndHashCode.Include
    @Column(name = "notification_id", length = 10, nullable = false)
    private String notificationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @ToString.Exclude
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "title", nullable = false)
    private String title;

    @Lob
    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(name = "typeNote", nullable = false, columnDefinition = "ENUM('ORDER_UPDATE','NEW_PET','VACCINATION_REMINDER','PROMOTION')")
    private NotificationType typeNote;

    @Column(name = "is_read", columnDefinition = "TINYINT(1) DEFAULT 0")
    private Boolean isRead;

    @CreationTimestamp
    @Column(name = "created_at", columnDefinition = "DATETIME(6) DEFAULT CURRENT_TIMESTAMP(6)", updatable = false)
    private LocalDateTime createdAt;
}

File: PetShop-petopia/client/src/server/routers/notification.ts
import { z } from "zod";
import { router, publicProcedure } from "../trpc";

export type Notification = {
  notification_id: string;
  user_id: string;
  title: string;
  content: string;
  typeNote: "ORDER_UPDATE" | "NEW_PET" | "VACCINATION_REMINDER" | "PROMOTION";
  is_read: boolean;
  created_at: string;
};

// Fake data demo, thay bằng truy vấn DB thực tế
const notifications: Notification[] = [
  {
    notification_id: "N001",
    user_id: "U001",
    title: "Đơn hàng đã được xác nhận",
    content: "Đơn hàng #1234 của bạn đã được xác nhận.",
    typeNote: "ORDER_UPDATE",
    is_read: false,
    created_at: new Date().toISOString(),
  },
  {
    notification_id: "N002",
    user_id: "U002",
    title: "Có thú cưng mới!",
    content: "Chúng tôi vừa thêm một bé mèo mới siêu dễ thương.",
    typeNote: "NEW_PET",
    is_read: false,
    created_at: new Date().toISOString(),
  },
];

export const notificationRouter = router({
  getAll: publicProcedure.query(() => {
    // Thay bằng truy vấn DB thực tế
    return notifications;
  }),
});
